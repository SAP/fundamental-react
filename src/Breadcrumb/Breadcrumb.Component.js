import React from 'react';
import { Breadcrumb, BreadcrumbItem } from '../';
import { Description, DocsText, DocsTile, Header, Import, Separator } from '../_playground';

export const BreadcrumbComponent = () => {
    const breadcrumbHrefCode = `<Breadcrumb>
    <BreadcrumbItem url="#" name="Link Text"/>
    <BreadcrumbItem url="#" name="Link Text"/>
    <BreadcrumbItem url="#" name="Link Text"/>
</Breadcrumb>`;

    const breadcrumbLinkCode = ` <Breadcrumb>
    <BreadcrumbItem link="/" name="Link Text" />
    <BreadcrumbItem link="/" name="Link Text" />
    <BreadcrumbItem link="/" name="Link Text" />
</Breadcrumb>`;

    return (
        <div>
            <Header>Breadcrumb</Header>
            <Description>
                The breadcrumb allows users to see the current page and navigation path to that page. Users can navigate
                to previous levels in the path. When clicking on the current page, a dropdown allows users to access
                other pages at that same level.
            </Description>
            <Import module='Breadcrumb, BreadcrumbItem' path='/fundamental-react/src/' />

            <Separator />

            <Description>An example using url (href attribute)</Description>
            <DocsTile>
                <Breadcrumb>
                    <BreadcrumbItem name='Link Text' url='#' />
                    <BreadcrumbItem name='Link Text' url='#' />
                    <BreadcrumbItem name='Link Text' url='#' />
                </Breadcrumb>
            </DocsTile>
            <DocsText>{breadcrumbHrefCode}</DocsText>

            <Separator />

            <Description>An example using link (routerLink)</Description>
            <DocsTile>
                <Breadcrumb>
                    <BreadcrumbItem link='/' name='Link Text' />
                    <BreadcrumbItem link='/' name='Link Text' />
                    <BreadcrumbItem link='/' name='Link Text' />
                </Breadcrumb>
            </DocsTile>
            <DocsText>{breadcrumbLinkCode}</DocsText>
        </div>
    );
};
