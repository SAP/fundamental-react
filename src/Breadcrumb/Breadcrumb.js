import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter, Link } from 'react-router-dom';

export const Breadcrumb = ({ children, ...props }) => {
    return <ul {...props} className='fd-breadcrumb'>{children}</ul>;
};

Breadcrumb.propTypes = {
    children: PropTypes.node
};

export const BreadcrumbItem = ({ url, link, name, className, ...props }) => {
    return (
        <BrowserRouter>
            <li className={`fd-breadcrumb__item${className ? ' ' + className : ''}`} {...props}>
                {link && <Link className='fd-breadcrumb__link' to={{ pathname: link }}>{name}</Link>}
                {url && <a className='fd-breadcrumb__link' href={url}>{name}</a>}
            </li>
        </BrowserRouter>
    );
};

BreadcrumbItem.propTypes = {
    name: PropTypes.string.isRequired,
    link: PropTypes.string,
    url: PropTypes.string
};
