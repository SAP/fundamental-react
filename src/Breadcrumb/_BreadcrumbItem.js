import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/breadcrumb.css';

const classnames = classnamesBind.bind(styles);

const BreadcrumbItem = ({ url, name, className, children, cssNamespace, ...props }) => {
    const renderLink = () => {
        if (!children && url) {
            return (
                <a
                    className={classnames(`${cssNamespace}-breadcrumb__link`)}
                    href={url}
                    tabIndex='0'>
                    {name}
                </a>
            );
        } else if (children) {
            return React.cloneElement(children, {
                'className': classnames(`${cssNamespace}-breadcrumb__link`),
                'tabIndex': '0'
            });
        }
    };

    const breadcrumbItemClasses = classnames(
        `${cssNamespace}-breadcrumb__item`,
        className
    );

    return (
        <li
            {...props}
            className={breadcrumbItemClasses}
            role='menuitem'
            tabIndex='-1'>
            {renderLink()}
        </li>
    );
};

BreadcrumbItem.displayName = 'Breadcrumb.Item';

BreadcrumbItem.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Text for the internal anchor tag */
    name: PropTypes.string,
    /** An anchor tag will be generated and set to the url prop. Name or child text must be provided */
    url: PropTypes.string
};

export default withStyles(BreadcrumbItem);
