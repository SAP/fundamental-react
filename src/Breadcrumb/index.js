export { default as Breadcrumb } from './Breadcrumb.js';
