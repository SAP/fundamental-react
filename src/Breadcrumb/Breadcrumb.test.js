import Breadcrumb from './Breadcrumb';
import { mount } from 'enzyme';
import React from 'react';

describe('<Breadcrumb />', () => {

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Breadcrumb component', () => {
            const element = mount(<Breadcrumb data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the BreadcrumbItem component', () => {
            const element = mount(<Breadcrumb.Item data-sample='Sample' name='Name' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <Breadcrumb ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('UL');
        expect(ref.current.className).toEqual('fd-breadcrumb');
    });
});
