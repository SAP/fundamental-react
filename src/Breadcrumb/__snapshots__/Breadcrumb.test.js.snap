// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> create breadcrumbs using router links 1`] = `
<ul
  className="fd-breadcrumb"
>
  <li
    className="fd-breadcrumb__item blue"
  />
  <li
    className="fd-breadcrumb__item"
  >
    <a
      className="fd-breadcrumb__link"
      href="/"
      onClick={[Function]}
    >
      Link Text
    </a>
  </li>
  <li
    className="fd-breadcrumb__item"
  >
    <a
      className="fd-breadcrumb__link"
      href="/"
      onClick={[Function]}
    >
      Link Text
    </a>
  </li>
</ul>
`;

exports[`<Breadcrumb /> create default breadcrumbs 1`] = `
<ul
  className="fd-breadcrumb"
>
  <li
    className="fd-breadcrumb__item"
  >
    <a
      className="fd-breadcrumb__link"
      href="#"
    >
      Link Text
    </a>
  </li>
  <li
    className="fd-breadcrumb__item"
  >
    <a
      className="fd-breadcrumb__link"
      href="#"
    >
      Link Text
    </a>
  </li>
  <li
    className="fd-breadcrumb__item"
  />
</ul>
`;
